web/cobrands/fixmystreet-uk-councils/alloy.js

```diff
@@ -9,19 +10,24 @@ OpenLayers.Protocol.Alloy = OpenLayers.Class(OpenLayers.Protocol.HTTP, {
         options = OpenLayers.Util.applyDefaults(options, this.options);
         var all_tiles = this.getTileRange_(options.scope.bounds, options.scope.layer.maxExtent, options.scope.layer.map);
         var rresp;
-        var start = new Date();
         var max = all_tiles.length;
-        options.scope.newRequest(start, max);
+        if (this.currentRequests.length) {
+            for (var j = 0; j < this.currentRequests.length; j++) {
+                this.currentRequests[j].priv.abort();
```

Comment:
This looks okay; however, as the last `OpenLayers.Protocol.Response` is returned by this read, `abort()` has already been called on that one by the `OpenLayers.Protocol.HTTP` abort function. Would it be better for this class to override `abort()` and do this abort loop there?

Reply:
I'm not sure I follow this.

Reply:
This `read` function returns the final `rresp` (and final entry in currentRequests) of the loop. OpenLayers.Strategy.BBOX stores that in `this.response` (in triggerRead). In a subsequent call to triggerRead, as `this.response` is set, it calls abort on the layer's protocol with that response: `this.layer.protocol.abort(this.response);` which is Protocol.HTTP which calls `response.priv.abort();` on it. It then gets to this read which will call abort() on all the currentRequests, including the one that has already had abort called on it.

Instead, if you overrode abort() in this class, that could do the loop and calling of al the priv.aborts() and read() wouldn't have to concern itself with aborting anything, which isn't really its job.